### PR TITLE
New module: installp for AIX packages

### PR DESCRIPTION
**Pull request by kairoaraujo**
##### SUMMARY
This module works with installp AIX packages.
With this module is possible to manage (install and remove) native
AIX packages using simple playbooks.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
module/packaging/os/installp

##### ANSIBLE VERSION
```
ansible 2.5.0 (os_installp_aix e7fd8f36f3) last updated 2017/09/12 21:42:57 (GMT +200)
  config file = None
  configured module search path = ['/Users/kairoaraujo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/kairoaraujo/Dev/aix_installp/ansible/lib/ansible
  executable location = /Users/kairoaraujo/Dev/aix_installp/ansible/bin/ansible
  python version = 3.6.2 (v3.6.2:5fd33b5926, Jul 16 2017, 20:11:06) [GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

##### ADDITIONAL INFORMATION
None

### CODE REVIEW

File: lib/ansible/modules/packaging/os/installp.py
Comment (Shaps) [inline lib/ansible/modules/packaging/os/installp.py]: I wouldn't call `module.params` as `packages` since it will hold also other parameters, not only the packages

File: lib/ansible/modules/packaging/os/installp.py
Comment (Shaps) [inline lib/ansible/modules/packaging/os/installp.py]: If you want to use a comma-separated list of package names, you should mention it in `DOCUMENTATION` and the `EXAMPLES`

**Review by Shaps — CHANGES_REQUESTED** [review]
Also, please squash your commits to remove the `Merge` commits, as the PR won't be merged otherwise.
https://github.com/ansible/ansible/blob/devel/CODING_GUIDELINES.md#git-practices

File: lib/ansible/modules/packaging/os/installp.py
Comment (abadger) [inline lib/ansible/modules/packaging/os/installp.py]: Rather than manually split here, you should define this as a list in the argument_spec:

``` python
            name=dict(aliases=["pkg"], required=True, type='list'),
```

An argument_spec list can either be handed a yaml list or it can receive a comma separated string (which it will automatically split into a list)

**Review by gforster — COMMENTED** [review]
curious on the status of this. looks like all is good?

**Comment by gforster** [comment]
bot_status

File: lib/ansible/modules/packaging/os/installp.py
Comment (gforster) [inline lib/ansible/modules/packaging/os/installp.py]:
```suggestion
version_added: "2.8"
```
Should this be re-targeted for 2.8?

**Comment by gforster** [comment]
Should this be renamed to aix_installp? I'm not sure that installp exists anywhere outside of AIX, but the aix_prefix may still be desired.

**Comment by bcoca** [comment]
note: unrelated to this module being accepted, we should add installp to fact gathering to identify ansible_pkg_mgr correctly.

**Comment by dagwieers** [comment]
> Should this be renamed to aix_installp? I'm not sure that installp exists anywhere outside of AIX, but the aix_prefix may still be desired.

It is not a requirement, most of the packaging modules are not prefixed. I do think it would be beneficial to move all AIX-specific modules to its own *aix/* sub-directory. But would only do it once we have everything merged and all related issues closed.

**Comment by dagwieers** [comment]
So I am going to fix a few things and merge this. It has been hold up for too long.
We will have to make integration tests for this module to make it easier for others to verify correct usage.

File: lib/ansible/modules/packaging/os/installp.py
Comment (dagwieers) [reply lib/ansible/modules/packaging/os/installp.py]: Fixed.

**Pull request merged**
